cloud/aws/providerconfig/types.go

```diff
@@ -14,25 +14,123 @@
 package providerconfig
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// AWSMachineProviderConfig is the type that will be embedded in a Machine.Spec.ProviderConfig field
```

Comment:
This doesn't actually say anything about what this struct is supposed to represent. Is it a single running instance, or template of an instance or what? The comment must clarify that, at the very least. Instead it talks about relationship between structs, which is something one can figure by reading the code anyway.

Reply:
Also, I think it's very reasonable to ask where this should be just an alias to [`ec2.Instance`](https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#Instance)?